### ISSUE TITLE
There is a jarring stutter only on first touch on menu items.

### ISSUE THREAD

**Issue by navdove** [root-two/react-native-drawer#51]
Only on first time, second time if any menu item is touched its fine. Please see attached gif.

While testing i also notice that the drawer after menu item select on first time, forces to stay open, even though i am calling close per the code below (handleSection function).

index.ios.js
  updateSelection(value){
    this.setState({currentView: this.state.views[value]});
    console.log('+++++++++was in updateSelection: '+this.state.currentView+' of '+this.state.views.length);
//    this.refs.drawer.close()
  },

ControlPanel.js
    handleSection: function(value) {
        console.log('calling handleSection with value ='+value)
        this.props.updateSelection(value);
        this.props.closeDrawer();
    },

```
render(){
return (
  <View style={styles.controlPanel}>
    <View style={{paddingTop:50, backgroundColor: 'teal'}}>
    <Image source={require('./img/logo.png')} style={{width: 100, height: 60, paddingTop: 40}}  >

    </Image>
        </View>
    <Button
      onPress={this.props.closeDrawer}
      text="Close Drawer"/>
      <TouchableHighlight onPress={() => { this.handleSection('0') }} >
          <View style={styles.drawerMenuItem}>
              <Icon name="navicon" size={30} color="teal" />
              <Text style={styles.drawerMenuItem}>
                  Tax and Audit
              </Text>

          </View>
      </TouchableHighlight>

      <TouchableHighlight onPress={() => { this.handleSection('1') }}>
          <View style={styles.drawerMenuItem}>
              <Icon name="navicon" size={30} color="teal" />
              <Text style={styles.drawerMenuItem}>
                  Latest Info
              </Text>

          </View>
      </TouchableHighlight>

      <TouchableHighlight onPress={() => { this.handleSection('2') }}>
          <View style={styles.drawerMenuItem}>
              <Icon name="navicon" size={30} color="teal" />
              <Text style={styles.drawerMenuItem}>
                  Innovation and Technology
              </Text>

          </View>
      </TouchableHighlight>
  </View>
)
```

![bug](https://cloud.githubusercontent.com/assets/5692908/11447135/1e0b7f76-950f-11e5-85db-5ad782c6e44c.gif)

**Comment by rt2zz** [comment]
can you paste your full setup including the drawer configuration

**Comment by navdove** [comment]
Sure.. here is index.ios.js , pretty much based on your example

var React = require('react-native');
var {
  AppRegistry,
  Text,
  View,
    StatusBarIOS
} = React;

var styles = require('./styles')
var drawerStyles = {
  drawer: {
    shadowColor: "#000000",
    shadowOpacity: 0.8,
    shadowRadius: 0,
  }
}

var Drawer = require('react-native-drawer')
var LatestView = require('./LatestView');
var MainView = require('./MyMainView');
var Innovation = require('./Innovation');
var MyControlPanel = require('./ControlPanel')

var deviceScreen = require('Dimensions').get('window')
var tweens = require('./tweens');

var counter = 0;

if (typeof window !== 'undefined') {
  window.React = React;
}

var iosApp = React.createClass({
  getInitialState(){
    var views = [<MainView  setParentState={this.setStateFrag}
                               openDrawer={this.openDrawer}
        />, <LatestView
        setParentState={this.setStateFrag}
        openDrawer={this.openDrawer}
        />, <Innovation
        setParentState={this.setStateFrag}
        openDrawer={this.openDrawer}
        />];

```
return {
  drawerType: 'overlay',
  openDrawerOffset:0,
  closedDrawerOffset:0,
  panOpenMask: .1,
  panCloseMask: .9,
  relativeDrag: false,
  panStartCompensation: true,
  openDrawerThreshold: .25,
  openDrawerOffset: .20,
  tweenHandlerOn: false,
  tweenEasing: 'linear',
  disabled: false,
  tweenHandlerPreset: null,
  acceptDoubleTap: true,
  acceptTap: false,
  acceptDoubleTap: false,
  acceptPan: true,
  rightSide: false,
  views: views,
  currentView:views[2]

}
```

  },

  setDrawerType(type){
    this.setState({
      drawerType: type
    })
  },

  tweenHandler(ratio){
    if(!this.state.tweenHandlerPreset){ return {} }
    return tweens[this.state.tweenHandlerPreset](ratio)
  },

  openDrawer(){
    this.refs.drawer.open()
  },

  setStateFrag(frag){
    this.setState(frag)
  },
  updateSelection(value){
    this.setState({currentView: this.state.views[value]});
//    this.refs.drawer.close()
  },

  render() {
    var controlPanel = <MyControlPanel closeDrawer={() => {this.refs.drawer.close()}} updateSelection={(val) => this.updateSelection(val)} />
    StatusBarIOS.setStyle('light-content', true);
    return (
      <Drawer
        ref="drawer"
        type={this.state.drawerType}
        animation={this.state.animation}
        openDrawerOffset={this.state.openDrawerOffset}
        closedDrawerOffset={this.state.closedDrawerOffset}
        panOpenMask={this.state.panOpenMask}
        panCloseMask={this.state.panCloseMask}
        relativeDrag={this.state.relativeDrag}
        panStartCompensation={this.state.panStartCompensation}
        openDrawerThreshold={this.state.openDrawerThreshold}
        content={controlPanel}
        styles={drawerStyles}
        disabled={this.state.disabled}
        tweenHandler={this.tweenHandler}
        tweenDuration={this.state.tweenDuration}
        tweenEasing={this.state.tweenEasing}
        acceptDoubleTap={this.state.acceptDoubleTap}
        acceptTap={this.state.acceptTap}
        acceptPan={this.state.acceptPan}
        changeVal={this.state.changeVal}
        side='left'
        >
        {this.state.currentView}
      </Drawer>
    );
  }
});

AppRegistry.registerComponent('iosApp', () => iosApp);

ControlPanel.js
var React = require('react-native');
var Icon = require('react-native-vector-icons/FontAwesome');
var {
  SwitchIOS,
  View,
  Text,
  Image,
    TouchableHighlight
} = React

var styles = require('./styles')
var Button = require('./Button')

module.exports = React.createClass({

```
handleSection: function(value) {
    console.log('calling handleSection with value ='+value)
    this.props.updateSelection(value);
```

//        this.props.closeDrawer();
    },

```
render(){
return (
  <View style={styles.controlPanel}>
    <View style={styles.menuHeader}>
        <Image source={require('./img/logo.png')} style={styles.menuHeaderLogo}  >
        </Image>
        <TouchableHighlight  onPress={this.props.closeDrawer} style={{alignSelf: 'flex-end'}}>
            <View style={styles.drawerMenuItem}>
            <Icon name="close" size={30} color="white"
                />
                </View>
        </TouchableHighlight>
    </View>
      <TouchableHighlight onPress={() => {this.props.closeDrawer(); this.handleSection('0') }} >
          <View style={styles.drawerMenuItem}>
              <Icon name="navicon" size={30} color="teal" />
              <Text style={styles.drawerMenuItem}>
                  Tax and Audit
              </Text>

          </View>
      </TouchableHighlight>

      <TouchableHighlight onPress={() => {this.props.closeDrawer(); this.handleSection('1') }}>
          <View style={styles.drawerMenuItem}>
              <Icon name="navicon" size={30} color="teal" />
              <Text style={styles.drawerMenuItem}>
                  Latest Info
              </Text>

          </View>
      </TouchableHighlight>

      <TouchableHighlight onPress={() => {this.props.closeDrawer(); this.handleSection('2') }}>
          <View style={styles.drawerMenuItem}>
              <Icon name="navicon" size={30} color="teal" />
              <Text style={styles.drawerMenuItem}>
                  Innovation and Technology
              </Text>

          </View>
      </TouchableHighlight>
      <TouchableHighlight onPress={ () => {this.props.closeDrawer()} }>
          <View style={styles.drawerMenuItem}>
              <Icon name="navicon" size={30} color="teal" />
              <Text style={styles.drawerMenuItem}>
                  Just Close Drawer
              </Text>

          </View>
      </TouchableHighlight>

  </View>
)
```

  }
})

Two  Views (LatestView.js and Innovation.js) like your MainView.js
var React = require('react-native');
var Icon = require('react-native-vector-icons/FontAwesome');

var {
  SwitchIOS,
  SliderIOS,
  PickerIOS,
  PickerItemIOS,
  View,
  ScrollView,
  Text,
  Component,
    TouchableHighlight
} = React

var styles = require('./styles')
var Button = require('./Button')

class LatestView extends Component{

```
toString() {
    return 'LatestView';
}

render(){
return (
  <ScrollView
    pointerEvents="box-none"
    style={styles.scrollView}
    scrollEventThrottle={200}
    contentInset={{top: 0}}
    >
    <View style={styles.toolbar}>
            <Icon name="navicon" size={30} color="white" onPress={this.props.openDrawer}
                />
            <Text style={styles.title}>
                Latest Info
            </Text>

        </View>

  </ScrollView>
)
```

  }
}

module.exports = LatestView;

# 

var React = require('react-native');
var Icon = require('react-native-vector-icons/FontAwesome');

var {
  SwitchIOS,
  SliderIOS,
  PickerIOS,
  PickerItemIOS,
  View,
  ScrollView,
  Text,
  Component,
    TouchableHighlight
} = React

var styles = require('./styles')
var Button = require('./Button')

class Innovation extends Component{

```
toString() {
    return 'Innovation';
}

render(){
return (
  <ScrollView
    pointerEvents="box-none"
    style={styles.scrollView}
    scrollEventThrottle={200}
    contentInset={{top: 0}}
    >
    <View style={styles.toolbar}>
            <Icon name="navicon" size={30} color="white" onPress={this.props.openDrawer}
                />
            <Text style={styles.title}>
                Innovation and Technology
            </Text>

        </View>

  </ScrollView>
)
```

  }
}

module.exports = Innovation;

and few similar views like this above view, that make my menu press targets. 

package.json
{
  "name": "iosApp",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "start": "react-native start"
  },
  "dependencies": {
    "react-native": "~0.15.0",
    "react-native-drawer": "~1.8.0",
    "react-native-vector-icons": "~0.8.5"
  }
}

On Second time tap, the menu goes away and no more stutters.

Thank you..

**Comment by navdove** [comment]
Hi Zack, did you get a change to reproduce my issue?

thanks-

**Comment by rt2zz** [comment]
@navdove figured out it was a bug with the viewport syncing. Should be fixed in `1.10.1`, let me know if that solves it for you

**Comment by navdove** [comment]
Thanks Zack, it works
